### ISSUE TITLE
FutureWarning: Conversion of the second argument of issubdtype from `int` to `np.signedinteger` is deprecated

### ISSUE THREAD

**Issue by Pastafarianist** [RaRe-Technologies/gensim#2188]
#### Description
```
/usr/lib/python3.7/site-packages/gensim/matutils.py:737: FutureWarning: Conversion of the second argument of issubdtype from `int` to `np.signedinteger` is deprecated. In future, it will be treated as `np.int64 == np.dtype(int).type`.
  if np.issubdtype(vec.dtype, np.int):
```

#### Steps/Code/Corpus to Reproduce

```
from gensim.test.utils import common_texts
from gensim.models import Word2Vec

model = Word2Vec(common_texts, size=100, window=5, min_count=1, workers=4)
print(model.wv.most_similar('human'))
```

#### Expected Results
No warning.

#### Actual Results
`FutureWarning`.

#### Versions
```
Linux-4.14.68-1-MANJARO-x86_64-with-arch-Manjaro-Linux
Python 3.7.0 (default, Jul 15 2018, 10:44:58) 
[GCC 8.1.1 20180531]
NumPy 1.15.1
SciPy 1.1.0
gensim 3.5.0
FAST_VERSION 0
```

#### Possible fix

Replace the offending line with

```
if np.issubdtype(vec.dtype, np.signedinteger) or np.issubdtype(vec.dtype, np.unsignedinteger):
```

Based on [Numpy docs](https://docs.scipy.org/doc/numpy-1.15.1/reference/arrays.dtypes.html?highlight=signedinteger#specifying-and-constructing-data-types). I haven't researched the issue thoroughly though and I'm not sure whether this is a proper fix.

**Comment by menshikh-iv** [comment]
@Pastafarianist thanks for the report, issue reproduced using mentioned dependencies.

**Comment by menshikh-iv** [comment]
@aquatiko 
> so here we should replace np.int by np.int64 to get the highest int type hierarchy 

no, for example, `int32` isn't "child type" for `int64`
```python
>>> isinstance(np.int32, np.int64)
False
```

**Comment by Esco0823** [comment]
Hello, I am still receiving this error. My versions are:
Python - 3.7
numpy - 1.15.4
gensim - 3.6.0

this is the code I am trying to run:
![image](https://user-images.githubusercontent.com/22353149/50310772-11535400-0471-11e9-86f3-f3d5de2baf06.png)

this is my output:
![image](https://user-images.githubusercontent.com/22353149/50310835-41025c00-0471-11e9-949f-a822225834c0.png)

 I am new to gensim and have very little python experience, so I'm not sure if this issue is on my end or not. It seems to be the same though. Any help is appreciated.

**Comment by JB0urne** [comment]
> 
> Hello, I am still receiving this error. My versions are:
> Python - 3.7
> numpy - 1.15.4
> gensim - 3.6.0
> 

Well, the Fix was implemented on 3rd Oct - the last release was 3.6.0 (September '18), so I think you have to wait for the next release-date ;)
